core/src/listener/listener.ts

```diff
@@ -215,30 +220,41 @@ function latestJob({
       throw e
     }
 
+    // happens only on localhost
     if (latestBlock < observedBlock) {
       logger.warn('latestBlock < observedBlock. Updating observed block to revert the condition.')
       observedBlock = Math.max(0, latestBlock - 1)
     }
 
-    const logPrefix = generateListenerLogPrefix(contractAddress, observedBlock, latestBlock)
+    let logPrefix = generateListenerLogPrefix(contractAddress, observedBlock, latestBlock)
     try {
       if (latestBlock > observedBlock) {
-        await redisClient.set(observedBlockRedisKey, latestBlock)
+        // create record for each block in unprocessed_blocks table
+        // dont include observedBlock as it's considered processed
+        await insertUnprocessedBlocks({
+          blocks: Array.from(
+            { length: latestBlock - observedBlock },
+            (_, i) => observedBlock + i + 1
+          ),
+          service
+        })
+
+        // Update observed_block table
+        await upsertObservedBlock({ service, blockNumber: latestBlock })
```

Comment:
do we have to wait for this db insertion? how about removing await keyword here?

Reply:
it should be safe assuming the upsert operation takes less than 0.5 seconds. I added await to make sure we don't keep repeating processing the same block in the latestJob. I'm a bit hesitant to assume upsert operation will always be under 0.5 seconds, what do you think?

Reply:
yh let's keep it as it is :)